tests/core/caching-utils/test_request_caching.py

```diff
@@ -187,9 +195,176 @@ def test_all_providers_do_not_cache_by_default_and_can_set_caching_properties(pr
+    "threshold",
+    (RequestCacheValidationThreshold.FINALIZED, RequestCacheValidationThreshold.SAFE),
+)
+@pytest.mark.parametrize("endpoint", BLOCKNUM_IN_PARAMS | BLOCKNUM_IN_RESULT)
+@pytest.mark.parametrize(
+    "blocknum,should_cache",
+    (
+        ("0x0", True),
+        ("0x1", True),
+        ("0x2", True),
+        ("0x3", False),
+        ("0x4", False),
+        ("0x5", False),
+    ),
+)
+def test_blocknum_validation_against_validation_threshold_when_caching(
+    threshold, endpoint, blocknum, should_cache, request_mocker
+):
+    w3 = Web3(
+        HTTPProvider(
+            cache_allowed_requests=True, request_cache_validation_threshold=threshold
+        )
+    )
+    with request_mocker(
+        w3,
+        mock_results={
+            endpoint: (
+                # mock the result to requests that return blocks
+                {"number": blocknum}
+                if "getBlock" in endpoint
+                # mock the result to requests that return transactions
+                else {"blockNumber": blocknum}
+            ),
+            "eth_getBlockByNumber": lambda _method, params: (
+                # mock the threshold block to be blocknum "0x2", return
+                # blocknum otherwise
+                {"number": "0x2"}
+                if params[0] == threshold.value
+                else {"number": params[0]}
+            ),
+        },
+    ):
+        assert len(w3.provider._request_cache.items()) == 0
+        w3.manager.request_blocking(endpoint, [blocknum, False])
+        assert len(w3.provider._request_cache.items()) == int(should_cache)
+
+
+@pytest.mark.parametrize(
+    "threshold",
+    (RequestCacheValidationThreshold.FINALIZED, RequestCacheValidationThreshold.SAFE),
+)
+@pytest.mark.parametrize("endpoint", BLOCKNUM_IN_PARAMS)
+@pytest.mark.parametrize(
+    "block_id,blocknum,should_cache",
+    (
+        ("earliest", "0x0", True),
+        ("earliest", "0x2", True),
+        ("finalized", "0x2", False),
+        ("safe", "0x2", False),
+        ("latest", "0x2", False),
+        ("pending", None, False),
+    ),
+)
+def test_block_id_param_caching(
+    threshold, endpoint, block_id, blocknum, should_cache, request_mocker
+):
+    w3 = Web3(
+        HTTPProvider(
+            cache_allowed_requests=True, request_cache_validation_threshold=threshold
+        )
+    )
+    with request_mocker(
+        w3,
+        mock_results={
+            endpoint: "0x0",
+            "eth_getBlockByNumber": lambda _method, params: (
+                # mock the threshold block to be blocknum "0x2" for all test cases
+                {"number": "0x2"}
+                if params[0] == threshold.value
+                else {"number": blocknum}
+            ),
+        },
+    ):
+        assert len(w3.provider._request_cache.items()) == 0
+        w3.manager.request_blocking(RPCEndpoint(endpoint), [block_id, False])
+        assert len(w3.provider._request_cache.items()) == int(should_cache)
```

Comment:
Nit: The assert here is a little awkward with `should_cache` as an int. Would suggest the ternary as you have in the other test below. Take or leave, slight readability improvement.
```suggestion
        cached_items = len(w3.provider._request_cache.items())
        assert cached_items == 1 if should_cache else cached_items == 0
```